reqhandler/request_handler.go

```diff
@@ -21,3 +22,34 @@ func readRequest(udpConn net.PacketConn) ([]byte, error) {
 	_, _, err := udpConn.ReadFrom(b)
 	return b, err
 }
+
+func extractID(d []byte) uint16 {
+	return (uint16(d[0]) << 8) | uint16(d[1])
```

Comment:
It should be done at a higher level, I've opened an issue for that. But there are a bunch of error checks we should do in addition to this.